tiledb/sm/query/sparse_global_order_reader.cc

```diff
@@ -1409,6 +1409,10 @@ Status SparseGlobalOrderReader::process_slabs(
         // Adjust the offsets buffer and make sure all data fits.
         var_buffer_size = compute_var_size_offsets<OffType>(
             stats_, result_cell_slabs, cell_offsets, query_buffer);
+        if (result_cell_slabs.empty()) {
```

Comment:
Shouldn't we use the incomplete + incomplete reason for this instead of an error?

Reply:
We return an error in the same case of Sparse unordered with duplicates, so I imitated that behavior. I will fix it for both in case we decide to go for incomplete reason.
@KiterLuc thoughts?

Reply:
Reverted it to return an Incomplete query as before and added tests that check on the reason returned.

I fixed the behavior to be similar also for `SparseUnorderedWithDuplicates` reader, that today returns an error instead of INCOMPLETE, but that change is breaking backwards compatibility. How should I handle this?  @Shelnutt2 @KiterLuc

Reply:
Discussed in sprint review and decided to proceed with this change.